File: example_reduce.cpp
#include <iostream>
#include <torch/torch.h>
#include "reduce.cpp"

int main() {
  int64_t n{10};
  torch::manual_seed(0);
  torch::Tensor X{torch::randn({n, 3})};
  torch::Tensor X_keys{torch::randint(2, {n, 2})};
  int64_t col{1};
  torch::Tensor out{reduce(X, X_keys, col)};
  std::cout << "X\n" << X << std::endl;
  std::cout << "X_keys\n" << X_keys << std::endl;
  std::cout << "col " << col << std::endl;
  std::cout << std::endl;
  std::cout << "reduced X\n" << out << std::endl;
  return 0;
}

File: example_reduce.py
# This file shows how to install the cpp reduce
# Note you have to do `pip install .` before using this script

import torch

try:
    import reduce_cpp
except ModuleNotFoundError as e:
    print("!!! Please install module with `pip install -e .` !!!")
    raise e

torch.manual_seed(0)
n = 10
X = torch.rand((n, 3))
X_keys = torch.randint(2, (n, 2))
col = 1
X_reduced = reduce_cpp.reduce(X, X_keys, col)
print("X\n", X)
print("X_keys\n", X_keys)
print("col", col)
print("X_reduced\n", X_reduced)
